index.js

```diff
@@ -3,14 +3,31 @@ const pathToBrowserBuild = require.resolve('happo-e2e/browser.build.js');
 
 const controller = new Controller();
 
+async function lazyLoadBrowserBundle(page) {
+  if (
+    await page.evaluate(() => typeof window.happoTakeDOMSnapshot === 'undefined')
+  ) {
+    await page.addScriptTag({ path: pathToBrowserBuild });
+
+    // Add timeout check for happoTakeDOMSnapshot
+    try {
+      await page.waitForFunction(
+        () => typeof window.happoTakeDOMSnapshot !== 'undefined',
+        { timeout: 10000 },
+      );
+    } catch (error) {
+      throw new Error('Timed out waiting for happoTakeDOMSnapshot to be defined');
+    }
+  }
+}
+
 module.exports = {
-  async init(contextOrPage) {
-    await contextOrPage.addInitScript({ path: pathToBrowserBuild });
-    await contextOrPage.addInitScript(`
-      window.addEventListener('load', () => {
-        window.happoTakeDOMSnapshot.init(window);
```

Comment:
I was wondering if we still needed to call this init function somewhere, but I found the function definition and discovered that it is now a noop, so removing this should be fine https://github.com/happo/happo-e2e/blob/5903e00c9dd2bb4d5a9cf9c2994181668ddfc634/takeDOMSnapshot.js#L347-L351

However, if it is possible for someone to have an updated version of happo-playwright but an old version of happo-e2e, they may run into a confusing situation. I'm not entirely sure what that experience would be like, but if we think it is likely that someone will run into this, then maybe we should make an adjustment to help them out.